app/models/spree/preference.rb

```diff
@@ -22,7 +22,8 @@ def value
         when :password
           self[:value].to_s
         when :decimal
-          BigDecimal(self[:value].to_s).round(2, BigDecimal::ROUND_HALF_UP)
+          BigDecimal(self[:value].to_s, exception: false)&.round(2, BigDecimal::ROUND_HALF_UP) ||
+            self[:value]
```

Comment:
Okay, I had a look through the codebase and decimal seems to be only used with calculators which you added validation for. :heavy_check_mark: 

And I hope that we won't build anything new around these Spree::Preferences because they are a pain to work with (as you've seen with the cache problem).